cafebabel/static/js/textarea-change.js

```diff
@@ -0,0 +1,27 @@
+/* Auto-expand textarea when typing http://jsfiddle.net/hmelenok/WM6Gq/ */
+function textAreaAutoResize(element, event, handler) {
+  element.addEventListener(event, handler, false)
+}
+
+function textAreaResizing() {
+  Array.from(document.querySelectorAll('textarea')).forEach(textArea => {
+    function resize() {
+      textArea.style.height = `${textArea.scrollHeight}px`
+    }
+
+    /* 0-timeout to get the already changed text */
+    function delayedResize() {
+      window.setTimeout(resize, 0)
+    }
+
+    textAreaAutoResize(textArea, 'change', resize)
+    textAreaAutoResize(textArea, 'cut', delayedResize)
+    textAreaAutoResize(textArea, 'paste', delayedResize)
+    textAreaAutoResize(textArea, 'drop', delayedResize)
+    textAreaAutoResize(textArea, 'keydown', delayedResize)
+
+    resize()
+  })
+}
+
+textAreaResizing()
```

Comment:
I didn't see this one line should be here but I guess it make sense too :)

Reply:
well since `textAreaResizing` is a function, it doesn't call it itself, does it? So we have to call it at one point

Reply:
A common pattern in JS is to use an [IIFE](https://en.wikipedia.org/wiki/Immediately-invoked_function_expression)

Reply:
Most of the purpose of IIFEs may have been replaced with the local scope of ES6 (that's one reason of using `const`).
Therefore we could have the code directly outside of a function but I guess it it more readable this way.
I had in mind to call `textAreaResizing()` from the main _scripts.js_ file but it doesn't feel ackward to me as such.
Any thought?

Reply:
It would mean that `textarea-change.js` should be included everywhere `scripts.js` is in use?

Reply:
I don't think we need `texarea-change.js` each time we need `script.js`. I am even pretty sure of it.